SETUP/upgrade/15/20201229_replace_project_templates.php

```diff
@@ -0,0 +1,48 @@
+<?php
+$relPath='../../../pinc/';
+include_once($relPath.'base.inc');
+
+header('Content-type: text/plain');
+
+// ------------------------------------------------------------
+
+echo "Replacing template references in project comments with their contents...\n";
+
+$sql = "
+    SELECT projectid, comments
+    FROM projects
+    WHERE comments like '%[template=%'
+";
+
+$result = mysqli_query(DPDatabase::get_connection(), $sql) or die( mysqli_error(DPDatabase::get_connection()) );
+
+while(list($projectid, $comments) = mysqli_fetch_row($result))
+{
+    // handle all templates in the comment
+    while(preg_match("/\[template=([^\]]+)\]/", $comments, $matches))
+    {
+        $template = $matches[1];
+        $template_path = "$relPath/templates/comment_files/$template";
+        if(!is_file($template_path))
+        {
+            die("Template '$template' referenced in $projectid but does not exist on the filesystem.\n");
```

Comment:
Maybe skip instead?

Reply:
Yes, that would be a better idea. The script could be run again after the problem is fixed and if the admin decides not to fix a problem it wouldn't break the system. Done.